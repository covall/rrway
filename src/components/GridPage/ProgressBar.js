import React from 'react';

const ProgressBar = ({ progress }) => {
  return (
    <div style={{ position: "relative" }}>
      <div style={{ width: "100%", height: 10, backgroundColor: "#AEAEAE" }} />
      <div
        style={{
          width: `${progress}%`,
          height: 10,
          backgroundColor: "green",
          position: "absolute",
          top: 0,
          transition: 'width 0.2s'
        }}
      />
    </div>
  );
};

export default ProgressBar;