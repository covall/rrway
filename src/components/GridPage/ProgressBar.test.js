import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import ProgressBar from './ProgressBar';


it('renders correctly', () => {
  const tree = renderer.create(
    <ProgressBar progress={25} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});