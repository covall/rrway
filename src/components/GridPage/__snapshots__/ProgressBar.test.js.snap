exports[`test renders correctly 1`] = `
<div
  style={
    Object {
      "position": "relative",
    }
  }>
  <div
    style={
      Object {
        "backgroundColor": "#AEAEAE",
        "height": 10,
        "width": "100%",
      }
    } />
  <div
    style={
      Object {
        "backgroundColor": "green",
        "height": 10,
        "position": "absolute",
        "top": 0,
        "transition": "width 0.2s",
        "width": "25%",
      }
    } />
</div>
`;
